src/hclust.jl

```diff
@@ -316,6 +316,137 @@ function hclust2{T<:Real}(d::Symmetric{T}, method::Function)
     hcat(mr[o], mc[o]), h[o]
 end
 
+# Ward linkage algorithm, using a Lance-Williams formula.
+#
+# Based on the R-coded version of the Fortran function hclust.f, originally
```

Comment:
What is the license of that file? We can only include code which is licensed under MIT or a compatible license.

Reply:
hclust.f is distributed with R:
https://github.com/wch/r-source/blob/trunk/src/library/stats/src/hclust.f
While there is no licence described on that file itself, the "surrounding" code is under the GPL. Do you think that this is a problem?

Thanks for your detailed comments on the code.

Reply:
Argh. Yes, that's a showstopper for inclusion into this package, as work derived from GPL code is itself licensed under the GPL, which is not compatible with the MIT license. So there are three options:
1) Relicense the package to GPL (other files can remain MIT, but the full package must be GPL). I'm afraid it's original authors wouldn't be fans of this solution.
2) Ship the code in a separate package. This of course has the drawback that it would be harder to discover. Note that  @davidavdav's HClust.jl, which has been merged into this package, was GPL, but I guess he didn't take inspiration from the R code. You could still revive that package and add your function there.
3) Get the authorization from the authors to relicense the code as MIT. Since that's not directly their code and Julia is open source, they might be open to this solution. You could contact Martin Maechler, I think he's discussed with Julia developers in the recent years. The two only other listed authors are Fritz Leisch and Ross Ihaka, so they could be contacted without too much work too.

If none of these possibilities works, somebody will have to rewrite the code only based on descriptions of the algorithms which are publically available. You could describe in detail what the R code does, as long as the person who write the code doesn't copy it directly. Yet's, that's really painful...

Reply:
For my original `HClust.jl`, I did not look at any code, just at the scientific articles quoted in the code.  I changed the license implicitly with the PR that included the code in this repository.

Reply:
Thanks. I should have really thought about these licensing issues before...
Is it generally sufficient to get permissions from one of the authors of the original code (say, the package maintainer), or do I need to get authorization from all of them?

Reply:
From all of them... :-/ At least we're lucky, the number of contributors is quite small (compare with e.g. this repo).

Reply:
I've had experience before asking for a change of licence from an original author with `MFCC.jl`, and that was quite easy.  But that was just a single author.

Reply:
@jmboehm Have you tried contacting the authors in the end?

Reply:
Yes. The original author, Fionn Murtagh, seemed initially very positive about it, but had some questions about the MIT license. When I explained it to him, he stopped responding. So I guess that means no. I'll fork my PR into a separate (GPL'ed) repo.

Reply:
Probably worth asking once again? We're all very busy, and it's easy to forget to reply to a message when you have no direct interest in it. You could also try to contact other authors, as it's a convincing argument to say "your coauthors have already accepted".

Reply:
I'll ask again, and will re-open if I get a positive answer.